### PR TITLE
refactor: move auth to http module

### LINKED ISSUE #6402: refactor: move util/got/auth to util/http/*
Waiting on #6401

### PR DESCRIPTION
**Pull request by rarkins**
Closes #6402

### CODE REVIEW

**Comment by rarkins** [comment]
I dare you to attempt to simplify it further without breaking lots of tests. I think we need to do #6083 next

File: lib/util/got/index.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
- */
-export const api = mergeInstances(got, auth);
+const dummy = create({
+  options: {},
+  handler: (options, next) => {
+    return next(options);
+  },
+});
+
+export const api = mergeInstances(got, dummy);
```
Comment (viceice) [inline lib/util/got/index.ts:13]: What happens when we simply directly use got? Again lots of failing tests?

Comment (rarkins) [reply lib/util/got/index.ts:13]: Yep. Actually lots of tests mock util/got so they immediately fail. I tried changing them over but it was dozens of test suites failures either way

**Review by viceice — APPROVED** [review]
Ok, so switch to mock first.

**Pull request merged**